### PR TITLE
Fix buildx add --provenance=false for pause image build and windows servercore cache

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup
/kind failing-test

#### What this PR does / why we need it:

https://testgrid.k8s.io/sig-k8s-infra-gcb#kubernetes-e2e-windows-servercore-cache
https://testgrid.k8s.io/sig-k8s-infra-gcb#post-kubernetes-push-image-pause. ([#119741](https://github.com/kubernetes/kubernetes/pull/119741)) only changed cloudbuild.yaml image. )

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubernetes/issues/121786
- following up of https://github.com/kubernetes/kubernetes/pull/121665 fixed a similar problem for etcd build

#### Special notes for your reviewer:

> gcr.io/k8s-staging-e2e-test-images/windows-servercore-cache:1.0-linux-amd64-1809 is a manifest list
make: *** [Makefile:43: all-build-and-push] Error 1
ERROR
ERROR: build step 0 "gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20230623-56e06d7c18" failed: step exited with non-zero status: 2

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by r0b2g1t — APPROVED** [review]
Good work.

**Comment by jsturtevant** [comment]
@marosset had a similiar change (https://github.com/kubernetes/kubernetes/pull/118991), but included `--sbom=false`.  just the changes here worked for the ETCD image?

/cc @marosset

**Comment by marosset** [comment]
let's add the --sbom=false too

File: build/pause/Makefile [hunk 1]
```diff
@@ -122,12 +122,12 @@ bin/wincat-windows-${ARCH}: windows/wincat/wincat.go
 
 container: .container-${OS}-$(ARCH)
 .container-linux-$(ARCH): bin/$(BIN)-$(OS)-$(ARCH)
-	docker buildx build --pull --output=type=${OUTPUT_TYPE} --platform ${OS}/$(ARCH) \
+	docker buildx build --provenance=false --sbom=false --pull --output=type=${OUTPUT_TYPE} --platform ${OS}/$(ARCH) \
```
Comment (BenTheElder) [inline build/pause/Makefile:125]: what version does this require?

Comment (humblec) [reply build/pause/Makefile:125]: @pacoxu do you have the docker buildx version handy where these flags were introduced ? in older OSs with older version of docker , the builds are failing..

**Comment by humblec** [comment]
@pacoxu do you have the docker buildx version handy where these flags were introduced ? in older OSs with older version of docker , the builds are failing..

**Comment by BenTheElder** [comment]
> we need folks to be able to build images from k/k repo, but that is broken without this PR. so landing this ( cc @kubernetes/release-team )


seems like we also could have fixed this by rolling back windows image version?

The flip side is we now require some unspecified newer docker version, in the future can we at least include the now required version in the release notes if nothing else?

**Pull request merged**